runbotics/runbotics-scheduler/src/scheduler-database/process/process-crud.service.ts

```diff
@@ -179,6 +183,16 @@ export class ProcessCrudService {
             id: In(updateDiagramDto.globalVariableIds),
         });
 
+        const allProcessCredentials = await this.processCredentialService.findAllByProcessId(process.id, user);
+
+        const credentialTypes = await findProcessActionTemplates(updateDiagramDto.definition);
+
+        allProcessCredentials.forEach(credential => {
+            if (!credentialTypes.includes(credential.credential.template.name)) {
+                this.processCredentialService.delete(credential.id, user);
+            }
+        });
```

Comment:
Shouldn't it be async operation? I would close this inside `await Promise.all()`